### PR TITLE
fix(WebSocketShard#destroy): wait for close and cleanup listeners

### PR DESCRIPTION
**Pull request by didinele**
**Please describe the changes this PR makes and why it should be merged:**
This is yet another attempt at fixing the exact same problem as #8399 and #8405

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating

### CODE REVIEW

File: packages/ws/src/ws/WebSocketShard.ts [hunk 1]
```diff
@@ -186,7 +186,18 @@ export class WebSocketShard extends AsyncEventEmitter<WebSocketShardEventsMap> {
 			this.connection &&
 			(this.connection.readyState === WebSocket.OPEN || this.connection.readyState === WebSocket.CONNECTING)
 		) {
+			// No longer need to listen to messages
+			this.connection.removeAllListeners('message');
```
Comment (vladfrangu) [inline packages/ws/src/ws/WebSocketShard.ts:190]: Use .off

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: packages/ws/src/ws/WebSocketShard.ts (hunk 1 above)
Comment (didinele) [reply packages/ws/src/ws/WebSocketShard.ts:190]: Is there a good reason for this?

Comment (vladfrangu) [reply packages/ws/src/ws/WebSocketShard.ts:190]: Consistency

Comment (didinele) [reply packages/ws/src/ws/WebSocketShard.ts:190]: Hm, I wonder if I have to do `.off('message', this.onMessage`) or `.off('message', this.onMessage.bind(this))` to match how I originally set up the listener. I'll mess with it

**Review by SpaceEEC — APPROVED** [review]

**Review by iCrawl — APPROVED** [review]

**Pull request merged**